addons/nightvision/functions/fnc_onVisionModeChanged.sqf

```diff
@@ -0,0 +1,29 @@
+/*
+ * Author: BaerMitUmlaut
+ * Disables turning on NVGs while the player aims down his sight.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: New vision mode <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, 1] call ace_nightvision_fnc_onVisionModeChanged
+ *
+ * Public: No
+ */
+
+#include "script_component.hpp"
+
+params ["_unit", "_visionMode"];
+
+if (GVAR(disableNVGsWithSights) && {(hmd _unit) != ""}) then {
+    (assignedVehicleRole _unit) params ["_role", "_turretPath"];
```

Comment:
nope. `assignedVehicleRole` does not work as you expect when you have an AI group leader.
Use `[_unit] call EFUNC(common,getTurretIndex) in [vehicle _unit] EFUNC(common,getTurretsFFV)` or something like that.